### PR TITLE
Add filtering and sorting to PVs and Resources tables

### LINKED ISSUE #794: Resources and PVs tables in Plan Wizard should have filtering and sorting controls
I was going to add this as part of https://github.com/konveyor/mig-ui/pull/776 but decided to break it into its own issue since that PR was getting a little large.

See Vince's mockups: https://docs.google.com/presentation/d/1hlP8wjzsO_9c9ri2rIz7o2kuIQvmKCNUhoQxTnUkHr0/edit#slide=id.g7f12813776_1_1

I have an in-progress branch for this, assigning myself.

### PR DESCRIPTION
**Pull request by mturley**
Closes #794.

This PR adds filtering and sorting controls to the tables in steps 2 and 3 of the plan wizard.

I'm adding a new common component `FilterToolbar` based on the usage example of `DataToolbar` here: https://www.patternfly.org/v4/documentation/react/demos/filtertabledemo . That demo involves a ton of boilerplate, is written as a class component, and is very coupled to the specific filter categories in the demo. Since we already need this in two tables and it's not terribly complex to abstract out, I rewrote the demo with hooks as `FilterToolbar`, which can be used for arbitrary filter categories. I'm planning to contribute it back to PatternFly eventually if possible.

I also added two new common hooks, `useFilterState` and `useSortState`, to make it easier to manage the state for these things. I chain them together with `usePaginationState` so each table's items are first filtered, then sorted, then paginated, before being mapped into row objects and passed into `<Table>`. I have a small `useEffect` line after this chain which makes sure the pagination will reset to page 1 when the sort or filters change.

Reusing this stuff to add sorting/filtering to any tables in the future should be easy peasy.

# Before
Screenshot 2020-04-13 01 17 51
Screenshot 2020-04-13 01 18 12

# After
Screenshot 2020-04-13 01 14 58
Screenshot 2020-04-13 01 15 33

With a sort and a filter applied:
Screenshot 2020-04-13 01 16 20

### CODE REVIEW

**Comment by mturley** [comment]
@ibolton336 @eriknelson this one is ready to go if you have a chance to review

**Comment by ibolton336** [comment]
Screen Shot 2020-04-13 at 12 56 16 PM
looking awesome! - just one thing I noticed, when you hit the select all button when filtering is active, you are selecting all namespaces even the ones that aren't shown @mturley

**Comment by mturley** [comment]
Hmm, ok that's interesting. Thanks @ibolton336 I think I know how to fix that

**Comment by mturley** [comment]
@ibolton336 should be fixed now (sorry for the force-pushes, I messed with some types and then realized those changes would conflict with https://github.com/konveyor/mig-ui/pull/780 so I reverted them)

**Comment by mturley** [comment]
Rebased and resolved conflicts.

File: src/app/common/components/FilterToolbar/FilterToolbar.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
+import React, { useState } from 'react';
```
Comment (eriknelson) [inline src/app/common/components/FilterToolbar/FilterToolbar.tsx:1]: Love these reusable types :+1:

File: src/app/common/components/FilterToolbar/FilterToolbar.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+  const setFilterValue = (category: FilterCategory, newValue: FilterValue) =>
+    setFilterValues({ ...filterValues, [category.key]: newValue });
+
+  const currentFilterCategory = filterCategories.find(
+    category => category.key === currentCategoryKey
+  );
+
+  return (
+    <DataToolbar id="pv-table-filter-toolbar" clearAllFilters={() => setFilterValues({})}>
```
Comment (mturley) [inline src/app/common/components/FilterToolbar/FilterToolbar.tsx:74]: Oops, I left a PV-table-specific ID in here. I'll fix in my next PR.

**Pull request merged**